pybossa/cache/users.py

```diff
@@ -366,9 +367,24 @@ def delete_taskbrowse_bookmarks(user):
     delete_memoized(get_user_by_id, user.id)
 
 
-def get_user_preferences(user_id):
+def get_user_preferences(user_id, map_to_country_codes=False):
     user = get_user_by_id(user_id)
     user_pref = user.user_pref or {} if user else {}
+    if map_to_country_codes and 'locations' in user_pref:
+        new_locations_set = set()
+        for location in user_pref['locations']:
+            new_locations_set.add(location)
+            # location is a country code, fetch country name
+            if len(location) == 2:
+                mapped_loc = app_settings.upref_mdata.get_country_by_country_code(location)
+            # location is a country name, fetch country code
+            else:
+                mapped_loc = app_settings.upref_mdata.get_country_code_by_country(location)
+            if mapped_loc is not None:
```

Comment:
It may be useful to log if an invalid country code is specified:

```python
if mapped_loc is not None:
  new_locations_set.add(mapped_loc)
else:
  logger.warning(f"Invalid country code '{location}' for user {user_id} in get_user_preferences")
```

Reply:
added logging via 6bbed61